xc/device.ml

```diff
@@ -1750,38 +1750,40 @@ module Vusb = struct
 
   let vusb_plug ~xs ~privileged ~domid ~id ~hostbus ~hostport ~version ~speed =
     debug "vusb_plug: plug VUSB device %s" id;
-    let usb_bus0 = "usb-bus.0" in
-    let ehci0    = "ehci.0" in
-    let speed_of_float x = if x <= 0. then
-                             `Unknown
-                           else if x <= 1.5 then
-                             `Low    (* v1.0 *)
-                           else if x <= 12. then
-                             `Full   (* v1.1 *)
-                           else if x <= 480. then
-                             `High   (* v2.0 *)
-                           else if x <= 5000. then
-                             `Super  (* v3.0 *)
-                           else
-                             `Unknown
-    in
     let get_bus () =
+      let usb_bus0 = "usb-bus.0" in
+      let ehci0 () =
+        vusb_controller_plug ~xs ~domid ~driver:"usb-ehci" ~driver_id:"ehci";
```

Comment:
Could this qemu message be sent just inside the `if Qemu.is_running ~xs domid then`?

I'd prefer separation between the preparation without side-effects and the interaction with qemu.

and idea would be to pass around a tuple of the bus and prepare_bus `(string * unit -> unit)` that can be executed afterwards.